autoload/ale/c.vim

```diff
@@ -91,3 +92,27 @@ function! ale#c#FindCompileCommands(buffer) abort
 
     return ''
 endfunction
+
+" Given a buffer number, return the relevant command from
+" compile_commands.json. Return 0 if none could be found.
+function! ale#c#GetCompileCommand(buffer) abort
```

Comment:
I recommend writing a function which doesn't do any I/O and returns the commands that are parsed, given a base directory and a List of lines in the file. Then it will be easy to write a series of unit tests against that function, and a small number of integration tests.

Reply:
Could, I thought you'd get relative paths from the file, and have to resolve them using a base directory.